rewards/snapshot/chain_snapshot.py

```diff
@@ -41,6 +41,34 @@ def chain_snapshot(chain: Network, block: int) -> Dict[str, Snapshot]:
     return balances_by_sett
 
 
+def weighted_sett_snapshot(
+        chain: Network, start_block: int, end_block: int,
+        sett: str, blacklist: bool, number_of_snapshots: int,
+) -> Snapshot:
+    """
+    Take a weighted snapshot of a sett between two blocks
+    """
+    assert end_block > start_block
+    snapshot = sett_snapshot(chain, start_block, sett, blacklist)
+    snapshot += sett_snapshot(chain, end_block, sett, blacklist)
+    rate = int((end_block - start_block) / number_of_snapshots)
+    # If rate == 0 it means that number of snapshots is too big, and it cannot be calculated
+    # properly. So, just divide snapshots balance values by 2(start and endblock and return)
+    # For ex: start block 13710328, end block 13710338 and num of snaps == 14
+    if rate == 0:
+        snapshot.balances = {k: v / 2 for k, v in snapshot.balances.items()}
```

Comment:
Technically this is not needed, due to the fact that the percentage that each user has of the snapshot will not change. This is because the way we calculate the rewards for each sett is based on the percentage of how much each user has of the snapshot